acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,94 @@
+
+  step 'On master, create new environments that will result in Failed' do
+    site_manifest = File.join(environmentpath, tmp_environment, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  fail('intended failure')
+}
+SITEPP
+    site_manifest2 = File.join(environmentpath, tmp_environment2, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest2, <<-SITEPP)
+node default {
+  exec{'/bin/false':}
+}
+SITEPP
+  end
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent),
+                         pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  target_identities = []
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment}" do
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'),
+               "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal(tmp_environment, action_result['environment'],
+                     "Result of pxp-module-puppet run on #{agent} should run with the "\
+                     "#{tmp_environment} environment")
+        assert_equal('failed', action_result['status'],
+                     "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: "\
+                     "'#{action_result["status"]}'")
+      end
+    end
+  end
+
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment2}" do
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment2,
```

Comment:
I have 2.40.0

From looking at beaker's code; I'm suspicious that when I try this code snippet:

    2.1.1 :001 > require 'tempfile'
     => true
    2.1.1 :002 > file = Tempfile.new('foo')
     => #<Tempfile:/var/folders/86/dcqmwc991h50ksqbq_l7z5400000gn/T/foo20160505-87693-mgtbka>
    2.1.1 :003 > file.path
     => "/var/folders/86/dcqmwc991h50ksqbq_l7z5400000gn/T/foo20160505-87693-mgtbka"

I get these permissions:

    Jamess-MacBook-Pro:~ jstocks$ ls -ltra /var/folders/86/dcqmwc991h50ksqbq_l7z5400000gn/T/foo20160505-87693-mgtbka
    -rw-------  1 jstocks  staff  0  5 May 10:51 /var/folders/86/dcqmwc991h50ksqbq_l7z5400000gn/T/foo20160505-87693-mgtbka

In case it was just my Macbook; I also tried on the shell box:

    james.stocks@shell:~$ ls -ltr /tmp/foo20160505-4223-a5q33g
    -rw------- 1 james.stocks allstaff 0 May  5 03:07 /tmp/foo20160505-4223-a5q33g

Would you mind trying the same snippet and seeing if you get the same or more permissive permissions? I think beyond this; beaker just uses net-scp with no options (so I don't see an opportunity to get inconsistent results).

If this continues to be a problem (for me); could we include a chmod command so it always works; then have a ticket for me to investigate; fix and tidy up?